mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -61,37 +62,53 @@
 @Slf4j
 class ExecutorStateManagerImpl implements ExecutorStateManager {
     private final Map<TaskExecutorID, TaskExecutorState> taskExecutorStateMap = new HashMap<>();
-    Cache<String, JobRequirements> pendingJobRequests = CacheBuilder.newBuilder()
+
+    /**
+     * Cache the available executors ready to accept assignments. Note these executors' state are not strongly
+     * synchronized and requires state level check when matching.
+     */
```

Comment:
Why is strong synchronization needed if it's only used by the actor?

Reply:
this was a prev comment. let me update it.